src/ocean/text/arguments/DateIntervalArgs.d

```
@@ -0,0 +1,436 @@
+
+        Params:
+            args = The arguments to process
+
+    ***************************************************************************/
+
+    public void processArgs ( Arguments args )
+    {
+        if ( args("time-range").set && args["time-range"].assigned.length == 2 )
+        {
+            cstring begin = args["time-range"].assigned[0];
+            cstring end = args["time-range"].assigned[1];
+
+            this.cli_begin = this.getlong(begin, 0);
+            this.cli_end = this.getlong(end, 24 * 3600 - 1);
+
+            if ( isTimeInterval(begin) && isTimeInterval(end) )
+            {
+                this.cli_begin = time(null) - parseTimeInterval(begin);
+                this.cli_end = time(null) + parseTimeInterval(end);
+            }
+
+            if ( isTimeInterval(begin) && this.cli_begin == 0 )
+            {
+                this.cli_begin = this.cli_end - parseTimeInterval(begin);
+            }
+
+            if ( isTimeInterval(end) && this.cli_end == 0 )
+            {
+                this.cli_end = this.cli_begin + parseTimeInterval(end);
+            }
+        }
+
+        if ( this.cli_begin == 0 && this.cli_end == 0 )
+        {
+            this.cli_begin = time(null) - this.default_interval;
+        }
+
+        if ( this.cli_begin == 0 )
+        {
+            this.cli_begin = time(null);
+        }
+
+        if ( this.cli_end == 0 )
+        {
+            this.cli_end = time(null);
+        }
+    }
+
+    private long getlong ( cstring value, long date_time )
+    {
+        if ( value == "now" )
+        {
+            return time(null);
+        }
+
+        long timestamp;
+        if ( toLong(value, timestamp) )
+        {
+            return timestamp;
+        }
+
+        long result;
+        DateConversion dummy_conv;
+        if ( timeToUnixTime(value, result, dummy_conv) )
+        {
+            result += date_time;
+        }
+
+        return result;
```

This will silently do nothing in case neither of conversions work. You probably want to throw some exception.